src/gc.c

```diff
@@ -1402,7 +1397,9 @@ static jl_taggedvalue_t **gc_sweep_page(jl_gc_pool_t *p, jl_gc_pagemeta_t **allo
         jl_taggedvalue_t **pfl_begin = NULL;
         while ((char*)v <= lim) {
             int bits = v->bits.gc;
-            if (!gc_marked(bits)) {
+            // if an object is past `lim_newpages` then we can guarantee it's garbage
+            if (!gc_marked(bits) || (char*)v >= lim_newpages) {
+                v->bits.gc = GC_CLEAN;
```

Comment:
```suggestion
                v->next = NULL; // v is now the tail of the freelist and will be assigned via `*pfl` later
```

Reply:
Why is this necessary? Don't we already null out the terminal pointers in https://github.com/JuliaLang/julia/blob/734cafabbfb1040aac7a92682ae78e8a6db699cf/src/gc.c#L1570?

Reply:
It is better than the RMW for setting GC_CLEAN, though indeed, this line of code is not needed.
```suggestion
```